mapper/tedge_mapper/tests/c8y_threshold_packet_size.rs

```diff
@@ -0,0 +1,150 @@
+        } else {
+            return Ok(());
+        }
+    }
+    Ok(())
+}
+
+async fn start_broker_local(cfile: &str) -> anyhow::Result<()> {
+    let config: Config = confy::load_path(cfile)?;
+    let (mut router, _console, servers, _builder) = async_locallink::construct_broker(config);
+    let router = tokio::task::spawn_blocking(move || -> anyhow::Result<()> { Ok(router.start()?) });
+    servers.await;
+    let _ = router.await;
+    Ok(())
+}
+
+async fn publish_big_message_wait_for_error() -> Result<(), anyhow::Error> {
+    // create a 20KB message
+    let buffer = create_packet(1024 * 20);
+
+    let topic = Topic::new("tedge/measurements")?;
+
+    let publish_client = Client::connect(
+        "publish_big_data",
+        &mqtt_client::Config::default().with_port(5885),
+    )
+    .await?;
+
+    let message = Message::new(&topic, buffer.clone()).qos(QoS::AtMostOnce);
+
+    let publish_handle = publish_client.publish(message);
+
+    // wait for error else timeout
+    let timeout = tokio::time::timeout(
+        std::time::Duration::from_secs(1),
+        subscribe_errors(&publish_client).map_err(|_s| TestJoinError::TestThresholdError),
+    )
+    .map_err(|_e| TestJoinError::ElapseTime);
+
+    // wait until one of the future returns error
+    let res = tokio::try_join!(
+        timeout,
+        publish_handle.map_err(|_e| TestJoinError::TestThresholdError)
+    );
+
+    match res {
+        Ok((first, _second)) => match first {
+            Err(TestJoinError::TestThresholdError) => {
+                return Ok(());
+            }
+            _ => {
+                anyhow::bail!("Did not catch error correctly");
+            }
+        },
+        _ => {
+            anyhow::bail!("test failed");
+        }
+    }
+}
+
+fn create_packet(size: usize) -> String {
+    let data: String = "Some data!".into();
+    let loops = size / data.len();
+    let mut buffer = String::with_capacity(size);
+    for _ in 0..loops {
+        buffer.push_str("Some data!");
+    }
+    buffer
+}
+
+fn setup() {
+    // set the port
+    let _tedge_output = Command::new("sudo")
+        .args(&["tedge", "config", "set", "mqtt.port", "5885"])
+        .output()
+        .expect("failed to execute process");
+
+    // start the tedge mapper
+    let _sysctl_output = Command::new("sudo")
+        .args(&["systemctl", "start", "tedge-mapper-c8y.service"])
+        .output()
+        .expect("failed to execute process");
+}
+
+fn cleanup() {
+    dbg!("cleanup");
+    // revert to default port
+    let _tedge_output = Command::new("sudo")
+        .args(&["tedge", "config", "unset", "mqtt.port"])
+        .output()
+        .expect("failed to execute process");
+
+    // stop tedge mapper
+    let _sysctl_output = Command::new("sudo")
+        .args(&["systemctl", "stop", "tedge-mapper-c8y.service"])
+        .output()
+        .expect("failed to execute process");
+}
```

Comment:
I understand the need for the setup function, but I'm struggling here with the need to call `tedge cli` with sudo? Is there a problem to create a configuration structure by hand in this function?
`TedgeConfig` is what you are looking for.